python/api-stages-lambda/api_stages_lambda/api_stages_lambda_stack.py

```diff
@@ -0,0 +1,120 @@
+from aws_cdk import (
+    Stack,
+    aws_apigateway as apigw,
+    aws_lambda as _lambda,
+    aws_iam as iam,
+    Aws, CfnOutput
+)
+from constructs import Construct
+
+class ApiStagesLambdaStack(Stack):
+
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Define AWS Lambda function and aliases
+        function = _lambda.Function(self, "LambdaFunction",
+            runtime=_lambda.Runtime.PYTHON_3_8,
+            handler="lambda-handler.handler",
+            code=_lambda.Code.from_asset("lambda")
+        )
+        version = function.current_version
+
+        # Define the REST API
+        api = apigw.RestApi(self, "DeploymentStagesAPI",
+            rest_api_name="MultiStageApi",
+            deploy=False
+        )
+
+        # Define GET method for /stage_info
+        stage_info_resource = api.root.add_resource("stage_info")
+        stage_info_get_method = stage_info_resource.add_method("GET",
+            authorization_type=apigw.AuthorizationType.IAM,
+            integration=apigw.AwsIntegration(
+                service="lambda",
+                region=Aws.REGION,
+                proxy=True,
+                path=f"2015-03-31/functions/{function.function_arn}:${{stageVariables.lambdaAlias}}/invocations"
+            )
+        )
+
+        # Define Prod specific resources
+        prod_alias = _lambda.Alias(self, "ProdAlias",
+            alias_name="prod",
+            version=version
+        )
+
+        prod_deployment = apigw.Deployment(self, "ProdApiDeployment", api=api)    
+        prod_stage = apigw.Stage(self, "ProdApiStage",
+            deployment=prod_deployment,
+            stage_name="prod",
+            variables={
+                "lambdaAlias": prod_alias.alias_name
+            }
+        )
+
+        prod_alias.add_permission("ProdStageInvokeProdAliasPermission",
```

Comment:
Is there a reason this can't be `prod_alias.grant_invoke(...)`? 

If so, please document it here.

Reply:
Hello! `grant_invoke` is great for granting permissions to an IAM Role or Principal (or higher level construct if it's supported). In this case, this would be `apigateway.amazonaws.com` OR an execution role that API Gateway assumes to invoke the Lambda function. In this case, I chose to make the resource policy more restrictive which meant I didn't need an explicit execution role for the APIGW --> Lambda integration.

Let me know if there is something I'm missing here. The current list of higher level constructs which implement the `IGrantable` interface are documented [here](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-iam.IGrantable.html).

Reply:
A comment to that effect would be helpful here: The fact this creates a tighter restriction than the `grant_invoke` pattern is useful for customers to understand.

Reply:
Added a comment to describe the permission code and why I selected it over `grant_invoke`. Thanks so much for reviewing! Can this one be merged in now?